packages/azure/src/azure-openai-provider.ts

```diff
@@ -80,21 +90,45 @@ export function createAzure(
       fetch: options.fetch,
     });
 
+  const createCompletionModel = (
+    modelId: string,
+    settings: OpenAICompletionSettings = {},
+  ) =>
+    new OpenAICompletionLanguageModel(modelId, settings, {
+      provider: 'azure-openai.completion',
+      url: ({ path, modelId }) =>
+        `https://${getResourceName()}.openai.azure.com/openai/deployments/${modelId}${path}?api-version=2024-05-01-preview`,
+      compatibility: 'compatible',
+      headers: getHeaders,
+      fetch: options.fetch,
+    });
+
   const provider = function (
     deploymentId: string,
-    settings?: OpenAIChatSettings,
+    settings?: OpenAIChatSettings | OpenAICompletionSettings,
   ) {
     if (new.target) {
       throw new Error(
         'The Azure OpenAI model function cannot be called with the new keyword.',
       );
     }
 
+    if (
+      deploymentId === 'gpt-35-turbo-instruct' ||
+      deploymentId === 'gpt-3.5-turbo-instruct'
+    ) {
+      return createCompletionModel(
+        deploymentId,
+        settings as OpenAICompletionSettings,
+      );
+    }
+
```

Comment:
this does not make sense for azure afaik since you get to name your deployments

Reply:
Yeah I don't have a good solution for this. Is there a better way to handle this case, since the current logic to determine whether to run `createChatModel` or `createCompletionModel` is determined entirely on the `modelId`/`deploymentId`?

Reply:
I could add a `type` to the provider, that says either `completion` or `chat`, then choose `createCompletionModel` or `createChatModel` that way?